fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -776,175 +886,214 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
-
-	// Queuing pre-resolution commit processing
-	TEST(self->latestLocalCommitBatchResolving.get() < localBatchNumber - 1);
-	wait(self->latestLocalCommitBatchResolving.whenAtLeast(localBatchNumber-1));
-	state Future<Void> releaseDelay = delay(std::min(SERVER_KNOBS->MAX_PROXY_COMPUTE, batchOperations*self->commitComputePerOperation[latencyBucket]), TaskPriority::ProxyMasterVersionReply);
-
-	if (debugID.present())
-		g_traceBatch.addEvent("CommitDebug", debugID.get().first(), "MasterProxyServer.commitBatch.GettingCommitVersion");
+	GetCommitVersionRequest req(
+		pProxyCommitData->commitVersionRequestNumber++,
+		pProxyCommitData->mostRecentProcessedRequestNumber,
+		pProxyCommitData->dbgid
+	);
+	GetCommitVersionReply versionReply = wait(brokenPromiseToNever(
+		pProxyCommitData->master.getCommitVersion.getReply(
+			req, TaskPriority::ProxyMasterVersionReply
+		)
+	));
 
-	GetCommitVersionRequest req(self->commitVersionRequestNumber++, self->mostRecentProcessedRequestNumber, self->dbgid);
-	GetCommitVersionReply versionReply = wait( brokenPromiseToNever(self->master.getCommitVersion.getReply(req, TaskPriority::ProxyMasterVersionReply)) );
-	self->mostRecentProcessedRequestNumber = versionReply.requestNum;
+	pProxyCommitData->mostRecentProcessedRequestNumber = versionReply.requestNum;
 
-	self->stats.txnCommitVersionAssigned += trs.size();
-	self->stats.lastCommitVersionAssigned = versionReply.version;
+	pProxyCommitData->stats.txnCommitVersionAssigned += context.trs.size();
+	pProxyCommitData->stats.lastCommitVersionAssigned = versionReply.version;
 
-	state Version commitVersion = versionReply.version;
-	state Version prevVersion = versionReply.prevVersion;
+	context.commitVersion = versionReply.version;
+	context.prevVersion = versionReply.prevVersion;
 
 	for(auto it : versionReply.resolverChanges) {
-		auto rs = self->keyResolvers.modify(it.range);
+		auto rs = pProxyCommitData->keyResolvers.modify(it.range);
 		for(auto r = rs.begin(); r != rs.end(); ++r)
 			r->value().emplace_back(versionReply.resolverChangesVersion,it.dest);
 	}
 
-	//TraceEvent("ProxyGotVer", self->dbgid).detail("Commit", commitVersion).detail("Prev", prevVersion);
+	//TraceEvent("ProxyGotVer", pProxyContext->dbgid).detail("Commit", commitVersion).detail("Prev", prevVersion);
 
-	if (debugID.present())
-		g_traceBatch.addEvent("CommitDebug", debugID.get().first(), "MasterProxyServer.commitBatch.GotCommitVersion");
+	if (debugID.present()) {
+		g_traceBatch.addEvent(
+			"CommitDebug", debugID.get().first(),
+			"MasterProxyServer.commitBatch.GotCommitVersion"
+		);
+	}
+
+	return Void();
+}
 
-	ResolutionRequestBuilder requests( self, commitVersion, prevVersion, self->version );
+ACTOR Future<Void> getResolution(CommitBatchContext* pContext) {
+	state CommitBatchContext& context = *pContext;
+	state ProxyCommitData* pProxyCommitData = pContext->pProxyCommitData;
+	state std::vector<CommitTransactionRequest>& trs = pContext->trs;
```

Comment:
No need to be declared as `state`, since it will not be used after `wait`.

Reply:
Yeah, I removed them.